sdk/src/Core/Amazon.Runtime/Credentials/FallbackCredentialsFactory.cs

```diff
@@ -124,61 +136,84 @@ public static AWSCredentials GetCredentials()
 
         public static AWSCredentials GetCredentials(bool fallbackToAnonymous)
         {
-            if (cachedCredentials != null)
-                return cachedCredentials;
-
-            List<Exception> errors = new List<Exception>();
-
-            foreach (CredentialsGenerator generator in CredentialsGenerators)
+            try
             {
-                try
```

Comment:
I don't think it is necessary to have a read lock here, It's not a problem if multiple threads are trying to read the credentials. So you can just leave it as it was before like:

```
if(cachedCredentials != null)
  return cachedCredentials;
```

Reply:
I meant to comment on line 141

Reply:
The problem is if a thread calls reset between the null check and return, the method will return null if we don't have a read lock